cirq-core/cirq/ops/measurement_gate.py

```diff
@@ -223,7 +223,7 @@ def __repr__(self):
         )
 
     def _value_equality_values_(self) -> Any:
-        return self.mkey, self.invert_mask, self._qid_shape
+        return str(self.mkey), self.invert_mask, self._qid_shape
```

Comment:
This propagates the `with_qubits` concern up a level:
```
ma = cirq.measure(q0, q1, key='0,1')
mb = cirq.measure(q0, q1)
ma == mb  # string keys are both "0,1"
ma.on(q1, q0) != mb.on(q1, q0)  # string keys are "0,1" and "1,0" respectively
```
Is there something that prevents us from using `self.mkey` directly here?

Reply:
Thanks for catching this! I agree that we shouldn't just push it up one level. 
Just something left over from between moving from old string based equality to the current full `MeasurementKey` checks.